test/assertEx.ts

```diff
@@ -0,0 +1,117 @@
+import * as assert from 'assert';
+import { parseError } from 'vscode-azureextensionui';
+
+function areUnorderedArraysEqual<T>(actual: T[], expected: T[]): { areEqual: boolean, message?: string } {
+    actual = actual.slice();
+    expected = expected.slice();
+    actual.sort();
+    expected.sort();
+
+    let message = `Actual:   ${JSON.stringify(actual)}\nExpected: ${JSON.stringify(expected)}`
+
+    if (!(actual.length === expected.length)) {
+        return { areEqual: false, message };
+    }
+
+    for (let i = 0; i < actual.length; ++i) {
+        if (actual[i] !== expected[i]) {
+            return { areEqual: false, message };
+        }
+    }
+
+    return { areEqual: true };
+}
+
+export function unorderedArraysEqual<T>(actual: T[], expected: T[], message?: string): void {
+    let result = areUnorderedArraysEqual(actual, expected);
+    assert(result.areEqual, `${message || "Unordered arrays are not equal"}\n${result.message}`);
+}
+
+export function notUnorderedArraysEqual<T>(actual: T[], expected: T[], message?: string): void {
+    let result = areUnorderedArraysEqual(actual, expected);
+    assert(!result.areEqual, `${message || "Unordered arrays are equal but were expected not to be"}\n${result.message}`);
+}
+
+export async function throwsOrRejectsAsync(block: () => Promise<any>, expected: {}, message?: string): Promise<void> {
+    let error: any;
+    try {
+        await block();
+    } catch (err) {
+        error = err;
+    }
+
+    if (!error) {
+        throw new Error(`Expected exception or rejection: ${parseError(expected).message}`);
+    }
+    for (let prop of Object.getOwnPropertyNames(expected)) {
+        assert.equal(error[prop], expected[prop], `Error did not have the expected value for property '${prop}'`);
+    }
+}
+
+suite("assertEx", () => {
```

Comment:
Can you add a little more description to this file? Not sure what 'Ex' stands for (Extensions? Extra?). I honestly thought these were just dummy tests at first and didn't realized these functions were primarily for use elsewhere

And usually the pattern is to have a `*.test.ts` to contain the actual tests. I know this is bit of an edge case, but it honestly confuses me that you're testing 'test' code and I think splitting this into `assertEx.ts` and `assertEx.test.ts` might help